src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp

```diff
@@ -279,7 +281,7 @@ void StorageObjectStorageQueue::shutdown(bool is_drop)
     {
         task->deactivate();
     }
-
+    std::lock_guard lock(startup_mutex);
```

Comment:
AFAIK we never call `shutdown` unless `startup` was successfully called

Reply:
So, it looks like here in other places since I've used ` TSA_GUARDED_BY(startup_mutex)` on `files_metadata`, it looks like we need to lock the mutex before accessing or modifying `files_metadata` everywhere - I had to add all these locks to make the builds pass. For example:
```
 error: reading variable 'files_metadata' requires holding mutex 'startup_mutex' [-Werror,-Wthread-safety-analysis]
971 |         files_metadata->alterSettings(changed_settings, local_context);
  ```

Reply:
We might probably not need `TSA_GUARDED_BY`  for `files_metadata` or `temp_metdata` so that will make it easier to just acquire the mutex where we want and not have thread safety analysis complain.

Reply:
>  since I've used  TSA_GUARDED_BY(startup_mutex)

let's remove it for now, TSA_GUARDED I mean, it overcomplicates things at the moment